diff --git a/build/soong/Android.bp b/build/soong/Android.bp
index 30a13198..544b5dbd 100644
--- a/build/soong/Android.bp
+++ b/build/soong/Android.bp
@@ -21,7 +21,7 @@ lineage_generator {
     name: "generated_kernel_includes",
 
     // The headers make command
-    cmd: "$(PATH_OVERRIDE_SOONG) $(KERNEL_MAKE_CMD) $(KERNEL_MAKE_FLAGS) -C $(TARGET_KERNEL_SOURCE) O=$(KERNEL_BUILD_OUT_PREFIX)$(genDir) ARCH=$(KERNEL_ARCH) $(KERNEL_CROSS_COMPILE) headers_install",
+    cmd: "./vendor/lineage/build/soong/generate_kernel_includes.sh",
 
     // Directories that can be imported by a cc_* module generated_headers property
     export_include_dirs: [
diff --git a/build/soong/generate_kernel_includes.sh b/build/soong/generate_kernel_includes.sh
new file mode 100644
index 00000000..43129e1e
--- /dev/null
+++ b/build/soong/generate_kernel_includes.sh
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+[ -x "$0" ] || chmod +x "$0"
+
+if [ -n "$KERNEL_MAKE_FLAGS" ]; then
+  "$PATH_OVERRIDE_SOONG" "$KERNEL_MAKE_CMD" "$KERNEL_MAKE_FLAGS" -C "$TARGET_KERNEL_HEADERS" O="$KERNEL_BUILD_OUT_PREFIX$genDir" ARCH="$KERNEL_ARCH" "$KERNEL_CROSS_COMPILE" headers_install
+else
+  exit 0
+fi
+
